services/blockchain-connector/shared/sdk/client.js

```diff
@@ -186,44 +130,83 @@ const invokeEndpoint = async (endpoint, params = {}, numRetries = NUM_REQUEST_RE
 	} while (retries--);
 };
 
-// Checks to ensure that the API Client is always alive
-if (config.isUseLiskIPCClient) {
-	const resetApiClientListener = async () => instantiateClient(true).catch(() => {});
-	Signals.get('resetApiClient').add(resetApiClientListener);
-} else {
-	let intervalTimeout;
-	const triggerRegularClientLivelinessChecks = intervalMs => {
-		intervalTimeout = setInterval(async () => {
-			const isAlive = await checkIsClientAlive();
-			if (!isAlive) instantiateClient(true).catch(() => {});
-		}, intervalMs);
-	};
-
-	const genesisBlockDownloadedListener = () => {
-		triggerRegularClientLivelinessChecks(CLIENT_ALIVE_ASSUMPTION_TIME_BEFORE_GENESIS);
-		logger.info(
-			`API client heartbeat checks scheduled every ${CLIENT_ALIVE_ASSUMPTION_TIME_BEFORE_GENESIS}ms. The frequency will be set to ${CLIENT_ALIVE_ASSUMPTION_TIME}ms after successful indexing of the genesis block.`,
-		);
-	};
-
-	const genesisBlockIndexedListener = indexStatus => {
-		if (
-			!isGenesisBlockIndexed &&
-			indexStatus.data &&
-			indexStatus.data.genesisHeight <= indexStatus.data.lastIndexedBlockHeight
-		) {
-			clearInterval(intervalTimeout);
-			triggerRegularClientLivelinessChecks(CLIENT_ALIVE_ASSUMPTION_TIME);
-			isGenesisBlockIndexed = true;
-			logger.info(
-				`API client heartbeat checks re-scheduled to run every ${CLIENT_ALIVE_ASSUMPTION_TIME}ms.`,
-			);
+const disconnectClient = async cachedClient => {
+	try {
+		await cachedClient.disconnect();
+	} catch (err) {
+		logger.warn(`Client disconnection failed due to: ${err.message}`);
+	}
+};
+
+const refreshClientsCache = async () => {
+	// Indicates if an active client was destroyed or no active clients available
+	let activeClientNotAvailable = cachedApiClients.length === 0;
+
+	// Check liveliness and remove non-responsive clients
+	let index = 0;
+	while (index < cachedApiClients.length) {
+		const cachedClient = cachedApiClients[index];
+		try {
+			if (!(await checkIsClientAlive(cachedClient))) {
+				cachedApiClients.splice(index, 1);
+				if (index === 0) {
+					activeClientNotAvailable = true;
+				}
+				await disconnectClient(cachedClient);
+			} else {
+				index++;
+			}
+		} catch (err) {
+			logger.info(`Failed to check client aliveness.\nError:${err.message}`);
+		}
+	}
+
+	// Initiate new clients if necessary
+	let missingClientCount = CONNECTION_LIMIT - cachedApiClients.length;
+
+	while (missingClientCount-- > 0) {
+		try {
+			await instantiateAndCacheClient();
+		} catch (err) {
+			logger.info(`Failed to instantiate new api client.\nError:${err.message}`);
 		}
```

Comment:
Should we update code to wait for clients to initialise? So that we don't end up with 0 missing clients 
```suggestion
	while (missingClientCount > 0) {
		try {
			await instantiateAndCacheClient();
			missingClientCount--;
		} catch (err) {
			logger.info(`Failed to instantiate new api client.\nError:${err.message}`);
			// TODO: Add delay here
		}
```